parlai/agents/seq2seq/seq2seq.py

```diff
@@ -226,16 +228,18 @@ def _init_model(self, states=None):
             # set loaded states if applicable
             self.model.load_state_dict(states['model'])
 
-        if self.use_cuda:
-            self.model.cuda()
-
         if opt['embedding_type'].endswith('fixed'):
             print('Seq2seq: fixing embedding weights.')
             self.model.decoder.lt.weight.requires_grad = False
             self.model.encoder.lt.weight.requires_grad = False
             if opt['lookuptable'] in ['dec_out', 'all']:
                 self.model.decoder.e2s.weight.requires_grad = False
 
+        if self.use_cuda:
+            self.model.cuda()
+            if self.multigpu:
+                self.model = torch.nn.DataParallel(self.model)
```

Comment:
you can do `self.model.encoder = self.model.module.encoder` (ditto for decoder), if you want to remove the `if multigpu` blocks later on

Reply:
doesn't quite work because beam_search (where we call model.encoder, etc.) is a static method. we could pass the encoder, etc. as an argument to beam search but i thought the if statements would end up being a bit cleaner

Reply:
fixed!